### PR TITLE
Fix #152: Use context=edit for WP.com REST post requests

### LINKED ISSUE #152: REST post requests should use context=edit
Like [iOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/75b3f4dc7503d07fb5222ce73d769e18f3eef507/WordPress/Classes/Networking/PostServiceRemoteREST.m#L74), we should make all post-related REST calls with `context=edit`. This returns posts in a format better suited for editing, which will also match the format of posts we receive from XML-RPC.

#### Default (visual) formatting:

```HTML
<p>Some content!</p>
<p>A new line.</p>
<p>Some <strong>bold</strong> text.</p>
<ul>
<li>item 1</li>
<li>item 2</li>
</ul>
<p>More text, <em>italics</em>.</p>
```

#### Formatting for editing (`context=edit`):

```HTML
Some content!

A new line.

Some <strong>bold</strong> text.
<ul>
	<li>item 1</li>
	<li>item 2</li>
</ul>
More text, <em>italics</em>.
```

Ref: https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/

### PR DESCRIPTION
**Pull request by aforcier**
Fixes #152, returning all post data in edit-friendly format, which also matches the format we receive post content in from XML-RPC.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java [hunk 1]
```diff
@@ -121,14 +128,16 @@ public void onErrorResponse(@NonNull BaseNetworkError error) {
...
-        final String url;
+        String url;
 
         if (post.isLocalDraft()) {
             url = WPCOMREST.sites.site(site.getSiteId()).posts.new_.getUrlV1_1();
         } else {
             url = WPCOMREST.sites.site(site.getSiteId()).posts.post(post.getRemotePostId()).getUrlV1_1();
         }
 
+        url += "?context=edit";
```
Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: We should use a utility method to add parameters, so we won't end up with 2 `?` in the url. Note: I'm thinking about the `lang=` parameter we must add to all requests.

Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: So when #155 is merged, we should replace this url concatenation by `request.addQueryParameter("context", "edit");`

Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/post/PostRestClient.java]: Done in ce5c7d3.

**Comment by aforcier** [comment]
Ready to go @maxme!

**Comment by maxme** [comment]
:shipit:

**Pull request merged**